File: Ejercicio 3/Ejercicio 3.cpp
#include <iostream>
using namespace std;

int main() {

    string NAME = "Renzo Leonardo Cienfuegos Cardenas";
    int GROUP = 2;

    int n;

    // Solicitamos la cantidad de elementos de los vectores
    cout << "Ingrese la cantidad de elementos de los vectores: ";
    cin >> n;

    int v[n], w[n];

    // Variable acumuladora para el producto escalar
    int producto = 0;

    // Solicitamos los elementos de ambos vectores
    for (int i = 0; i < n; i++) {

        cout << "Ingrese el elemento " << i + 1 << " del vector V: ";
        cin >> v[i];

        cout << "Ingrese el elemento " << i + 1 << " del vector W: ";
        cin >> w[i];

        // Acumulamos el producto de las componentes correspondientes
        producto += v[i] * w[i];

    }

    // Mostramos el resultado
    cout << "\nEl producto escalar de los vectores es: "
         << producto << endl;

    return 0;
}

File: Ejercicio 3/Ejercicio 3.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Solicitamos la cantidad de elementos de los vectores
n = int(input("Ingrese la cantidad de elementos de los vectores: "))

# Creamos los vectores vacíos
v = []
w = []

# Variable acumuladora para el producto escalar
producto = 0

# Solicitamos los elementos de ambos vectores
for i in range(n):

    elemento_v = int(input(f"Ingrese el elemento {i + 1} del vector V: "))
    v.append(elemento_v)

    elemento_w = int(input(f"Ingrese el elemento {i + 1} del vector W: "))
    w.append(elemento_w)

    # Acumulamos el producto de las componentes correspondientes
    producto = producto + (elemento_v * elemento_w)

# Mostramos el resultado
print(f"\nEl producto escalar de los vectores es: {producto}")
